### PR TITLE
Update docs for v1

### PR DESCRIPTION
**Pull request by restrry**
| Q                        | A  |
| ------------------------ | ---  |
| Fixed Issues?            | update docs |
| Patch: Bug Fix?          |    |
| Major: Breaking Change?  |    |
| Minor: New Feature?      |    |
| Tests Added + Pass?      | Yes |
| Any Dependency Changes?  |    |

1. could you check that I didn't miss something?
1. as I understand all task internal changes didn't affect public API. did it?
we removed `__is*` flags from `task` interface, but it doesn't mean to be a part of public interface.
1. also we added `mainTask` to `task` interface. seems it's a detail of internal implementation should we expose it at all?

### CODE REVIEW

**Comment by shinima** [comment]
We still need to use `task.mainTask.status` in `runCancelledEffect()`.  

A task may have its iterator consumed up, but still wait for a long-running child task. In such cases, `mainTask.status` is `DONE` and `task.status` is `CANCELLED`, we need `mainTask.status` to determine whether a saga is cancelled or not.

```javascript
function* saga() {
  yield fork(longRunningSaga)
  // mainTask.status is DONE here
  // but if the task is cancelled at this point, the task.status will be CANCELLED
}
```

**Comment by shinima** [comment]
3. The refactor to `task` and `mainTask` should be internal only. I did not change the public interface of Task. so we don't need to document the changes.

**Comment by Andarist** [comment]
`mainTask` is special anyway so maybe we should just transition it to CANCELLED state when the task gets cancelled?

**Comment by shinima** [comment]
`mainTask.status` should reflect the fact of running status of the iterator, so I think changing mainTask.status is not a good idea.

On the other hand, I think maybe using `cb(task.isCancelled())` in `runCancelledEffect()` is enough. `mainTask.status` becoming DONE means the main body of the saga has finished, so the saga will never yield a cancelled effect to query the cancellation status. That means we can simply ignore the cases when `mainTask.status` is `DONE`, and we do not need to care the edge case [that I just described](https://github.com/redux-saga/redux-saga/pull/1700#issuecomment-444439738).

However, I do a quick test against *using `cb(task.isCancelled())` in `runCancelledEffect()`* on my PC, two test cases are failing.

**Comment by Andarist** [comment]
I'm investigating switching to `task.isCancelled()` in `runCancelledEffect`. The problem is in order of things when propagating the cancellation. I'll report back when I will have more concrete findings or a solution for this.

File: packages/core/index.d.ts [hunk 1]
```diff
@@ -58,11 +58,14 @@ export interface SagaMiddleware<C = {}> extends Middleware {
 
   setContext(props: Partial<C>): void;
 }
+interface onErrorRuntime {
```
Comment (Andarist) [inline packages/core/index.d.ts]: I'd call it something like `ErrorInfo` or something, `onErrorRuntime` suggests to me that this is a handler function

Comment (restrry) [reply packages/core/index.d.ts]: ok, will fix after https://github.com/redux-saga/redux-saga/pull/1703 merged

**Comment by Andarist** [comment]
Could u rebase?

**Comment by restrry** [comment]
done

**Comment by Andarist** [comment]
❤️ gonna drop RC this week! Thanks for all the work you have put into the project
